atc/configwarning.go

```diff
@@ -15,7 +16,12 @@ var validIdentifiers = regexp.MustCompile(`^[\p{Ll}\p{Lt}\p{Lm}\p{Lo}][\p{Ll}\p{
 var startsWithLetter = regexp.MustCompile(`^[^\p{Ll}\p{Lt}\p{Lm}\p{Lo}]`)
 var invalidCharacter = regexp.MustCompile(`([^\p{Ll}\p{Lt}\p{Lm}\p{Lo}\d\-_.])`)
 
-func ValidateIdentifier(identifier string, context ...string) *ConfigWarning {
+func ValidateIdentifier(identifier string, context ...string) (*ConfigWarning, error) {
+	var err error
+	if identifier == "" {
+		return nil, errors.New("identifier cannot be an empty string")
```

Comment:
Perhaps this should incorporate the `context` as we do with warnings - that way you won't need all the `fmt.Sprintf("%s: %s", identifier, err.Error())` in `atc/configvalidate/validate.go`

```suggestion
		return nil, fmt.Errorf("%s: identifier cannot be an empty string", strings.Join(context, ""))
```